src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/HMACSHA3_256.cs

```diff
@@ -0,0 +1,390 @@
+namespace System.Security.Cryptography
+{
+    /// <summary>
+    /// Computes a Hash-based Message Authentication Code (HMAC) by using the SHA3-256 hash function.
+    /// </summary>
+    public class HMACSHA3_256 : HMAC
+    {
+        private HMACCommon _hMacCommon;
+        internal const int BlockSize = 136; // FIPS 202 Table 3.
+
+        /// <summary>
+        /// The hash size produced by the HMAC SHA3-256 algorithm, in bits.
+        /// </summary>
+        public const int HashSizeInBits = 256;
+
+        /// <summary>
+        /// The hash size produced by the HMAC SHA3-256 algorithm, in bytes.
+        /// </summary>
+        public const int HashSizeInBytes = HashSizeInBits / 8;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HMACSHA3_256" /> class with a randomly generated key.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// <see cref="HMACSHA3_256" /> is a type of keyed hash algorithm that is constructed from the SHA3-256 hash
+        /// function and used as a Hash-based Message Authentication Code (HMAC). The HMAC process mixes a secret key
+        /// with the message data, hashes the result with the hash function, mixes that hash value with the secret key
+        /// again, and then applies the hash function a second time. The output hash is 256 bits in length.
+        /// </para>
+        /// <para>
+        /// This constructor uses a 136-byte, randomly generated key.
+        /// </para>
+        /// </remarks>
+        public HMACSHA3_256()
+            : this(RandomNumberGenerator.GetBytes(BlockSize))
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HMACSHA3_256" /> class with the specified key data.
+        /// </summary>
+        /// <param name="key">
+        /// The secret key for <see cref="HMACSHA3_256" />. The key can be any length.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="key" /> is <see langword="null" />.
+        /// </exception>
+        public HMACSHA3_256(byte[] key)
+        {
+            ArgumentNullException.ThrowIfNull(key);
+            CheckSha3Support();
+
+            this.HashName = HashAlgorithmNames.SHA3_256;
+            _hMacCommon = new HMACCommon(HashAlgorithmNames.SHA3_256, key, BlockSize);
+            base.Key = _hMacCommon.ActualKey!;
+            BlockSizeValue = BlockSize;
+            HashSizeValue = HashSizeInBits;
+            Debug.Assert(HashSizeValue == _hMacCommon.HashSizeInBits);
+        }
+
+        /// <summary>
+        /// Gets a value that indicates whether the algorithm is supported on the current platform.
+        /// </summary>
+        /// <value>
+        /// <see langword="true" /> if the algorithm is supported; otherwise, <see langword="false" />.
+        /// </value>
+        public static bool IsSupported => HashProviderDispenser.MacSupported(HashAlgorithmNames.SHA3_256);
```

Comment:
I don't think they're generally expected to be runtime consts... more they're expected to be commensurate with the expect of the thing being noted as supported or not.  It matters a lot that `Vector128<T>.IsSupported` for example is a const because the operations it's guarded are measured in a very small number of cycles, plus it's important to be able to dead-code eliminate other code paths.  Here the operations being protected are relatively super expensive, and I suspect there's not a ton of trimming to be done.  If, however, we do think folks will have many different if/else blocks and we want to enable the JIT to trim away ones that won't be used, and/or if this is non-trivially expensive, we should cache it with ` { get; } = ` as you suggest.